pyximport/test/test_pyximport.py

```diff
@@ -68,5 +77,27 @@ def make_ext(name, filename):
     remove_tempdir(tempdir)
 
 
-if __name__=="__main__":
-    test()
+def test_zip():
+    try:
+        import test_zip_module
+    except ImportError:
+        pass
+    else:
+        assert False, "test_zip_module already exists"
+
+    fd, zip_path = tempfile.mkstemp(suffix=".zip")
```

Comment:
But the NTF would do the cleanup for you. No need for the try-finally in the end. (Besides, "handle" is not a good name for a variable that holds a file-like object.)